ros2_easy_test/env.py

```diff
@@ -351,3 +393,70 @@ def call_service(
 
         future = client.call_async(request)
         return cast(SrvTypeResponse, self.await_future(future, timeout=timeout_call))
+
+    def _get_action_objects(self, name: str, goal_class: Type) -> _ActionObjects:
+        """Returns the service client for the given service name."""
+
+        with self._registered_actions_lock:
+            # Get the type of the action.
+            # This is a bit tricky but relieves the user from passing it explicitly
+            module = import_module(goal_class.__module__)
+            # We cut away the trailing "_Goal" from the type name, which has length 5
+            base_type_name = goal_class.__name__[:-5]
+            base_type_class: Type = getattr(module, base_type_name)
+
+            objects = _ActionObjects(
+                client=ActionClient(self, base_type_class, name, callback_group=self._action_cb_group)
+            )
+            self._registered_actions[name] = objects
+            return objects
+
+    def send_action_goal_and_wait_for_response(
```

Comment:
I find myself in the need of an intermediate step: sending a goal and getting the goal handle. Such that I have the option to cancel it.

I might pick this up later, but for terminology: I would prefer `send_action_goal_and_wait_for_result`.
As the `response` of a goal actually communicates back the goal handle and if it was accepted or not. The `result` is the the result of completing of the task.

Pseudo code for a later enhancement:
```
    def send_action_goal_and_wait_for_result(args):
        goal_handle, response = send_action_goal_and_wait_for_result(args)
        result = wait_for(goal_handle)
        return goal_handle, feedbacks, result
```

_Please note, I'm only asking for a name change of the function at the moment :slightly_smiling_face:_

Reply:
Yup, makes sense!

Regarding the option for cancellation, It might even simplify the code since it splits up the main function a bit. Generally, though, I think this is one of the cases where users might have to use the env as a node and create their services themselves since I reckon it will rarely be used. Let's indeed discuss that in another Issue/PR. :)

Reply:
Updated the name.